From 3d640fef633614f9291583705bbec1b6d98ec3d0 Mon Sep 17 00:00:00 2001
From: Zheng Liu <i6122f@gmail.com>
Date: Wed, 23 Jun 2021 16:54:35 -0400
Subject: [PATCH] [riscv64] Fix IsIllegalInstruction function

Check whether the right most 16 bits are all-zero.

Change-Id: I13bb8856888cbabc19c1f9354048f05ff9e4aacb
Reviewed-on: https://chromium-review.googlesource.com/c/v8/v8/+/2983839
Reviewed-by: Ji Qiu <qiuji@iscas.ac.cn>
Commit-Queue: Ji Qiu <qiuji@iscas.ac.cn>
Cr-Commit-Position: refs/heads/master@{#75343}
---
 src/codegen/riscv64/constants-riscv64.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/codegen/riscv64/constants-riscv64.h b/src/codegen/riscv64/constants-riscv64.h
index 9527ab064d..bd1f63b673 100644
--- a/src/codegen/riscv64/constants-riscv64.h
+++ b/src/codegen/riscv64/constants-riscv64.h
@@ -738,7 +738,7 @@ class InstructionBase {
   };
 
   inline bool IsIllegalInstruction() const {
-    uint8_t FirstHalfWord = *reinterpret_cast<const uint16_t*>(this);
+    uint16_t FirstHalfWord = *reinterpret_cast<const uint16_t*>(this);
     return FirstHalfWord == 0;
   }
 
-- 
2.35.1

